### PR TITLE
fix worker&fuse options & worker tiredstore

### PR DESCRIPTION
**Pull request by zwwhdls**
### Ⅰ. Describe what this PR does
1. add options in JuiceFSRuntime.spec.fuse and get mount options of fuse pod from it instead of dataset.mount.option
2. get worker cache from tiredstore
3. cache-size&cache-dir in worker.options is deprecated, send event in runtime
4. fix cache capacity in dataset when cache-size is set in worker.options.

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/juicefs/cache.go [hunk 1]
```diff
@@ -20,9 +20,10 @@ import (
 	"fmt"
 	"strconv"
 
+	v1 "k8s.io/api/core/v1"
```
Comment (cheyang) [inline pkg/ddc/juicefs/cache.go:23]: Can you use `corev1 "k8s.io/api/core/v1"`?

File: pkg/ddc/juicefs/cache.go [hunk 2]
```diff
@@ -35,6 +36,20 @@ func (j *JuiceFSEngine) queryCacheStatus() (states cacheStates, err error) {
...
 		}
+		// if cacheSize is overwrited in worker options, deprecated
+		if j.runtime.Spec.Worker.Options["cache-size"] != "" {
+			// cacheSize is in MiB
+			c, e := strconv.Atoi(j.runtime.Spec.Worker.Options["cache-size"])
+			if e != nil {
+				err = e
+				return
+			}
+			cachesize, e = strconv.ParseUint(strconv.Itoa(c*1024*1024), 10, 64)
```
Comment (cheyang) [inline pkg/ddc/juicefs/cache.go:47]: How about using the following way:

```golang
if cacheSizeStr := j.runtime.Spec.Worker.Options["cache-size"]; cacheSizeStr != "" {
	cacheSizeMB, err := strconv.ParseUint(cacheSizeStr, 10, 64)
	if err != nil {
		return err
	}

	cachesize := cacheSizeMB * 1024 * 1024
}
```

File: pkg/ddc/juicefs/transform.go [hunk 3]
```diff
@@ -106,6 +107,31 @@ func (j *JuiceFSEngine) transformWorkers(runtime *datav1alpha1.JuiceFSRuntime, v
 		value.Worker.NodeSelector = runtime.Spec.Worker.NodeSelector
 	}
 
+	// options
+	mount := dataset.Spec.Mounts[0]
+	var tiredStoreLevel *datav1alpha1.Level
+	if len(runtime.Spec.TieredStore.Levels) != 0 {
+		tiredStoreLevel = &runtime.Spec.TieredStore.Levels[0]
```
Comment (cheyang) [inline pkg/ddc/juicefs/transform.go:114]: tiredStoreLevel -》 tieredStoreLevel

File: pkg/ddc/juicefs/transform.go [hunk 4]
```diff
@@ -129,6 +155,55 @@ func (j *JuiceFSEngine) transformWorkers(runtime *datav1alpha1.JuiceFSRuntime, v
...
+				}
+			}
+		}
+	}
+	if value.Edition == CommunityEdition {
+		if _, ok := workerOptionMap["metrics"]; !ok {
+			metricsPort := DefaultMetricsPort
+			if value.Fuse.MetricsPort != nil {
+				metricsPort = *value.Worker.MetricsPort
+			}
```
Comment (xliuqq) [inline pkg/ddc/juicefs/transform.go:184]: why use `Worker.MetricsPort` but judge `Fuse.MetricsPort != nil`？

Comment (zwwhdls) [reply pkg/ddc/juicefs/transform.go:184]: Mistake, thks

File: pkg/ddc/juicefs/cache.go [hunk 5]
```diff
@@ -20,25 +20,39 @@ import (
...
+		var cacheSizeMB uint64
+		cacheSizeMB, err = strconv.ParseUint(cacheSizeStr, 10, 64)
+		if err != nil {
 			return
 		}
+
+		// cacheSize is in MiB
+		cachesize = cacheSizeMB * 1024 * 1024
+	}
+	if cachesize != 0 {
```
[the remark this answers is not in the document]
Comment (zwwhdls) [reply pkg/ddc/juicefs/cache.go:51]: done

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**